src/librustdoc/html/static/js/externs.js

```diff
@@ -8,10 +8,30 @@ function initSearch(searchIndex){}
 
 /**
  * @typedef {{
- *   raw: string,
- *   query: string,
- *   type: string,
- *   id: string,
+ *     isExact: boolean,
+ *     name: string,
+ *     fullPath: Array<string>,
+ *     pathWithoutLast: Array<string>,
+ *     pathLast: string,
+ *     generics: Array<QueryElement>,
+ * }}
+ */
+var QueryElement;
+
+/**
+ * @typedef {{
+ *     original: string,
+ *     val: string,
```

Comment:
I see `val` in a few places and I don't understand what it means. Could you rename this to a more meaningful field / variable name?